learning_resources_search/api.py

```diff
@@ -477,6 +482,26 @@ def adjust_original_query_for_percolate(query):
     return order_params(query)
 
 
+def percolate_matches_for_document(document_id):
+    """
+    Percolate matching queries for a given learning resource
+    and call signal handler with matches
+    """
+    resource = LearningResource.objects.get(id=document_id)
+    index = get_default_alias_name(resource.resource_type)
+    search = Search()
+    results = search.query(
+        Percolate(field="query", index=index, id=str(document_id))
+    ).execute()
+    percolate_ids = [result.id for result in results.hits]
```

Comment:
The same code works fine on mitopen-rc, so I'm going to assume something is off with my local docker environment.

Reply:
spoke with matt. this may have been something local he was experiencing since i could not repro and it seemed to work on an rc deploy. marking as resolved.

Reply:
I ran `recreate_index --all` on RC last night and this morning I can reproduce the problem on RC.

1. enter a django shell: `heroku run python manage.py shell`
2. Enter the following code:
   ```python
    import pdb
    from opensearch_dsl import Search
    from opensearch_dsl.query import Percolate
    from learning_resources.models import LearningResource
    from learning_resources_search.connection import get_default_alias_name
    
    document_id=5154
    resource = LearningResource.objects.get(id=document_id)
    index = get_default_alias_name(resource.resource_type)
    search = Search()
    results = search.query(
        Percolate(field="query", index=index, id=str(document_id))
    ).execute()
    print(results.hits[0])
    print(results.hits[0].id)
```

Reply:
Probably best to use a serializer for this anyway

```python
serialized_data = PercolateQuerySerializer(instance=query).data
``` 

and then make any adjustments if necessary for the search index (like in `serialize_learning_resource_for_update`)

Reply:
Actually it looks like `serialize_percolate_query` is only used by `get_percolate_documents` which doesn't seem to be used, so maybe both of these functions can be deleted?

I think I found the culprit here:

```python
def serialize_bulk_percolators(ids):
    """
    Serialize percolators for bulk indexing

    Args:
        ids(list of int): List of percolator id's
    """
    for percolator in PercolateQuery.objects.filter(id__in=ids):
        yield {
            "_id": percolator.id,
            "query": {**remove_child_queries(percolator.query)},
        }
```

If I change "_id` to `"id"` then restart containers and recreate the index, I no longer get the error.

Would probably be better to have both `serialize_bulk_percolators` and `serialize_percolate_query_for_update` use the same serializer class to ensure they're consistent.

Reply:
For example, both `tasks.update_learning_resource` and  `serializers.serialize_learning_resource_for_bulk` call the same function `serializers.serialize_learning_resource_for_update` which uses the `LearningResourceSerializer` class and then makes a few search-specific modifications to the serialized data.

Reply:
Glad this was caught. I think there was an "id" -> "_id" mismatch oddity that came up due to the complexity of testing this end to end.

I pushed changes that consolidate the methods and made them both use the percolatequeryserializer. What I'm currently doing to test:
```
docker compose down
docker compose up
docker compose exec web /bin/bash
> ./manage.py recreate_index --all
> ./manage.py createsuperuser
```
login as super user then go to /api/v1/learning_resources_user_subscription/subscribe/ and post ```{"q":"test"}``` and ```{"q":"physics"}``` 

then:
```
> ./manage.py backpopulate_mitxonline_data
> ./manage.py backpopulate_prolearn_data
```
verify results in celery log.

then in django shell run:
```
from learning_resources_search.api import percolate_matches_for_document
percolate_matches_for_document(32)

```